trin-execution/src/walkers/trie_walker.rs

```diff
@@ -0,0 +1,378 @@
+                B256::from(U256::from(3)).as_slice(),
+                B256::from(U256::from(3)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(4)).as_slice(),
+                B256::from(U256::from(4)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(5)).as_slice(),
+                B256::from(U256::from(5)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(6)).as_slice(),
+                B256::from(U256::from(6)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(7)).as_slice(),
+                B256::from(U256::from(7)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(8)).as_slice(),
+                B256::from(U256::from(8)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(9)).as_slice(),
+                B256::from(U256::from(9)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(10)).as_slice(),
+                B256::from(U256::from(10)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(11)).as_slice(),
+                B256::from(U256::from(11)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(12)).as_slice(),
+                B256::from(U256::from(12)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(13)).as_slice(),
+                B256::from(U256::from(13)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(14)).as_slice(),
+                B256::from(U256::from(14)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(15)).as_slice(),
+                B256::from(U256::from(15)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(16)).as_slice(),
+                B256::from(U256::from(16)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(17)).as_slice(),
+                B256::from(U256::from(17)).as_slice(),
+            )
+            .unwrap();
+            trie.insert(
+                B256::from(U256::from(18)).as_slice(),
+                B256::from(U256::from(18)).as_slice(),
+            )
+            .unwrap();
+            trie.root_hash().unwrap();
+        }
+
+        let root_hash = trie.lock().root_hash().unwrap();
+        let walker = TrieWalker::new(root_hash, trie.lock().db.clone()).unwrap();
```

Comment:
Isn't `trie.lock().db.clone()` just `db`?

Reply:
It is an Arc of the traited object

Reply:
So it is Arc<DB>